assistant_dialog_skill_analysis/utils/lang_utils.py

```diff
@@ -0,0 +1,156 @@
+
+        if self.language_code == "de":
+            from spacy.lang.de import German
+
+            self.tokenizer = Tokenizer(German().vocab)
+            self.stemmer = SnowballStemmer(language="german")
+            self.stop_words = self.load_stop_words(
+                "assistant_dialog_skill_analysis/resources/de/stopwords"
+            )
+
+        if self.language_code == "it":
+            from spacy.lang.it import Italian
+
+            self.tokenizer = Tokenizer(Italian().vocab)
+            self.stemmer = SnowballStemmer(language="italian")
+            self.stop_words = self.load_stop_words(
+                "assistant_dialog_skill_analysis/resources/it/stopwords"
+            )
+
+        if self.language_code == "cs":
+            from spacy.lang.cs import Czech
+
+            self.tokenizer = Tokenizer(Czech().vocab)
+            self.stop_words = self.load_stop_words(
+                "assistant_dialog_skill_analysis/resources/cs/stopwords"
+            )
+
+        if self.language_code == "pt":
+            from spacy.lang.pt import Portuguese
+
+            self.tokenizer = Tokenizer(Portuguese().vocab)
+            self.stemmer = SnowballStemmer(language="portuguese")
+            self.stop_words = self.load_stop_words(
+                "assistant_dialog_skill_analysis/resources/pt/stopwords"
+            )
+
+        if self.language_code == "es":
+            from spacy.lang.es import Spanish
+
+            self.tokenizer = Tokenizer(Spanish().vocab)
+            self.stemmer = SnowballStemmer(language="spanish")
+            self.stop_words = self.load_stop_words(
+                "assistant_dialog_skill_analysis/resources/es/stopwords"
+            )
+
+    def preprocess(self, sentence):
+        sentence = sentence.lower()
+        sentence = self.strip_punctuations(sentence)
+        if self.language_code in ["fr", "es", "cs", "es", "pt"]:
+            sentence = self.accent_removal(sentence)
+
+        return sentence
+
+    def load_stop_words(self, path):
+        stopwords = set()
+        with open(path, "r", encoding="utf-8") as file:
+            for line in file:
+                line = line.strip()
+                stopwords.add(line)
+        return stopwords
+
+    def strip_punctuations(self, sentence):
+        """
+        function to strip punctuations from the utternace
+        :param utterance:
+        :return:
+        """
+
+        sentence = sentence.replace("'s", " is ")
+        sentence = re.sub(self.punctuation_pattern, " ", sentence)
+        return sentence
+
+    def accent_removal(self, sentence):
+        """from facebook research xlm preprocessing
+        https://github.com/facebookresearch/XLM"""
+        sentence = unicodedata.normalize("NFD", sentence)
+        output = []
+        for char in sentence:
+            cat = unicodedata.category(char)
+            if cat == "Mn":
+                continue
+            output.append(char)
+        return "".join(output)
```

Comment:
You could actually write the whole method as oneliner and it should be even faster:

```
return "".join(ch for ch in unicodedata.normalize("NFD", sentence) if unicodedata.category(ch) != "Mn")
```